modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/EditObjectFolder.tsx

```diff
@@ -139,34 +214,247 @@ export default function EditObjectFolder({
+							});
+						}}
+						helpMessage={Liferay.Language.get(
+							'unique-key-for-referencing-the-object-definition'
+						)}
+						onExternalReferenceCodeChange={(
+							externalReferenceCode: string
+						) => {
+							const updatedElements = elements.map((element) => {
+								if (
+									isNode(element) &&
+									(element as Node<ObjectDefinitionNodeData>)
+										.data?.id ===
+										selectedObjectDefinitionNode.data?.id
+								) {
+									return {
+										...element,
+										data: {
+											...element.data,
+											externalReferenceCode,
+										},
+									};
+								}
+
+								return element;
+							}) as Elements<ObjectDefinitionNodeData>;
+
+							dispatch({
+								payload: {
+									newElements: updatedElements,
+								},
+								type: TYPES.SET_ELEMENTS,
+							});
+						}}
+						onGetEntity={() =>
+							API.getObjectDefinitionById(
+								selectedObjectDefinitionNode.data?.id as number
+							)
+						}
+						saveURL={`/o/object-admin/v1.0/object-definitions/${selectedObjectDefinitionNode.data.id}`}
+					/>
+				)}
+
+			{modelBuilderModals.editObjectFolder && (
 				<ModalEditObjectFolder
 					externalReferenceCode={
 						selectedObjectFolder.externalReferenceCode
 					}
 					handleOnClose={() => {
-						setShowModal((previousState) => ({
-							...previousState,
-							editObjectFolder: false,
-						}));
+						dispatch({
+							payload: {
+								modelBuilderModals: {
+									...modelBuilderModals,
+									editObjectFolder: false,
+								},
+							},
+							type: TYPES.UPDATE_VISIBILITY_MODEL_BUILDER_MODALS,
+						});
 					}}
 					id={selectedObjectFolder.id}
 					initialLabel={selectedObjectFolder.label}
 					name={selectedObjectFolder.name}
 				/>
 			)}
 
-			{showModal.publishObjectDefinitions && (
+			{modelBuilderModals.publishObjectDefinitions && (
 				<ModalPublishObjectDefinitions
 					disableAutoClose={true}
 					dispatch={dispatch}
 					elements={elements}
 					handleOnClose={() => {
-						setShowModal((previousState) => ({
-							...previousState,
-							publishObjectDefinitions: false,
-						}));
+						dispatch({
+							payload: {
+								modelBuilderModals: {
+									...modelBuilderModals,
+									publishObjectDefinitions: false,
+								},
+							},
+							type: TYPES.UPDATE_VISIBILITY_MODEL_BUILDER_MODALS,
+						});
+					}}
+				/>
+			)}
+
+			{modelBuilderModals.redirectToEditObjectDefinitionDetails && (
+				<RedirectToEditObjectDetailsModal
+					handleOnClose={() => {
+						dispatch({
+							payload: {
+								modelBuilderModals: {
+									...modelBuilderModals,
+									redirectToEditObjectDefinitionDetails: false,
+								},
+							},
+							type: TYPES.UPDATE_VISIBILITY_MODEL_BUILDER_MODALS,
+						});
 					}}
+					viewObjectDetailsURL={formatActionURL(
+						editObjectDefinitionURL,
+						selectedObjectDefinitionNode?.data?.id ?? 0
```

Comment:
Hey Arthur, I had provided you with an [example](https://github.com/liferay/liferay-portal/commit/f91c213fad6a8ea89b17ebe4ceef3927cdc54d46#diff-0817cfb1b7931a909c4ef50d5c8cb8150cd75a4d22dfc7b7a43d3beb3e4026c3R430-R450) in the last review where I suggested that you use selectedObjectDefinitionNode?.data in the modal rendering validation. Could you make this change to avoid using the ?? 0

Reply:
done!